### ISSUE TITLE
HELP - Wireguard's inability to block censorship

### ISSUE THREAD

**Issue by rlrexyz** [donaldzou/WGDashboard#262]
First of all, I would like to say that I admire your project.
@donaldzou 

But I have a request from you. Can you create the same dashboard in shadowsocks/v2ray?

Your project is great, but wireguard gets stuck in internet censorship due to its structure, but shadowsocks/v2ray can bypass internet censorship.

When I try to integrate Shadowsocks into Wireguard, I have to download a separate program for shadowsocks to connect, etc. I am facing problems.
But let me state that I am not very experienced in these matters. I tried to integrate one of these two into wireguard instead of leaving wireguard and using shadowsocks/v2ray just so that I could use your wonderful dashboard, but I was not very successful. If there is a simpler way, I would be very happy if you could help me. All this effort is for me to be able to use this wonderful dashboard of yours...
Thanks...

Also, if anyone has a solution, I am waiting for your solutions.

**Comment by donaldzou** [comment]
Hi @rlrexyz, could you elaborate more on "wireguard gets stuck in internet censorship due to its structure" please? Unfortunately, currently I don't have any plans to port the dashboard for Shadowsocks/V2Ray as there're many wonderful dashboard out there already :)

**Comment by rlrexyz** [comment]
Hello @donaldzou 
Yes, there are dashboards for shadowsocks, but none of them are as beautiful as yours.

If we talk about Wireguard's censorship, for example, modems in schools, etc. In some places, devices are not allowed to connect to an outside IP address. That's why Wireguard cannot be used in such cases because its traffic is not private.

However, programs such as Shadowsocks/v2ray can be used without being exposed to any censorship because they hide the traffic.

Also, since I will not use my Wireguard server alone, I cannot turn to other alternative programs such as Shadowsocks/v2ray. The reason is that Wireguard is more useful. As an alternative, there is Tailscale, but it is not open source either. I'm stuck, your help would be great for me.

Thanks...

**Comment by donaldzou** [comment]
Ah I see, but in this case Shadowsocks/V2ray won't help much since your network blocked outside IP? Have you tried to run WireGuard on some other more frequently used ports such as 8080?

**Comment by donaldzou** [comment]
Also I'm thinking since WireGuard run on UDP, will it due to your network block UDP traffics?

**Comment by rlrexyz** [comment]
As a result, I have been thinking about this incident for more than a week, but I have not achieved any success.

But when I look at where I've come
 
-Wireguard does not hide traffic, but it is useful and has most of the features I was looking for.

-Shadowsocks/v2ray hides traffic and avoids censorship, but they do not have the features I want.

I only have one option left: passing Wireguard's traffic through Shadowsocks or v2ray, but doing this confused me.

Also, if I do this, I still haven't figured out whether the connected mobile device needs to download anything extra related to "Shadowsocks/v2ray". I would be very happy if you could enlighten me.

Thanks for your time.

**Comment by rlrexyz** [comment]
> Hi, I quickly Googled the issue and turns out someone else is also doing it: https://superuser.com/questions/1829695/forward-wireguard-traffic-through-shadowsocks

I tried many ways but I will try again, thank you. I'll get back to you after trying it...

**Comment by rlrexyz** [comment]
Hello again @donaldzou 

After a long struggle, I decided to leave wireguard. Both because it does not hide the traffic and because it does not provide any convenience for a solution despite having such a deficiency...

In my opinion, this is the biggest disadvantage of Wireguard. What is the point of a VPN if it does not hide the traffic? If we look at the point we have reached in the end, we have learned this with 2 weeks of experience.

Reluctantly, I installed X-UI and started using v2ray. (I already feel the lack of features, design beauty and convenience in your dashboard)

In conclusion, thank you for your interest. I hope that one day you will release a v2ray/shadowsocks compatible version of your Dashboard. (I think I will be the first person to use it :D)

Thank you for everything. I will continue to follow your project. I hope one day you decide to release a version for v2ray/shadowsocks :)
have a nice day...

**Comment by donaldzou** [comment]
> Hello again @donaldzou
> 
> After a long struggle, I decided to leave wireguard. Both because it does not hide the traffic and because it does not provide any convenience for a solution despite having such a deficiency...
> 
> In my opinion, this is the biggest disadvantage of Wireguard. What is the point of a VPN if it does not hide the traffic? If we look at the point we have reached in the end, we have learned this with 2 weeks of experience.
> 
> Reluctantly, I installed X-UI and started using v2ray. (I already feel the lack of features, design beauty and convenience in your dashboard)
> 
> In conclusion, thank you for your interest. I hope that one day you will release a v2ray/shadowsocks compatible version of your Dashboard. (I think I will be the first person to use it :D)
> 
> Thank you for everything. I will continue to follow your project. I hope one day you decide to release a version for v2ray/shadowsocks :) have a nice day...

Yeah I have to admit this could cause some issues fro some WireGuard users, since WireGuard is not designed to hide traffics.. Sad to see you're leaving but hope you found a solution that fits you :)
